ui_framework/src/components/form/field_search/field_search.js

```diff
@@ -3,28 +3,50 @@ import React, {
 } from 'react';
 import classNames from 'classnames';
 
+import {
+  KuiFormControlLayout,
+} from '../form_control_layout';
 
-export const KuiFieldSearch = ({ className, id, name, placeholder, value, ...rest }) => {
+import {
+  KuiValidatableControl,
+} from '../validatable_control';
+
+export const KuiFieldSearch = ({
+  className,
+  id,
+  name,
+  placeholder,
+  value,
+  isInvalid,
+  ...rest,
+}) => {
   const classes = classNames('kuiFieldSearch', className);
 
   return (
-    <input
-      type="search"
-      id={id}
-      name={name}
-      placeholder={placeholder}
-      className={classes}
-      value={value}
-      {...rest}
-    />
+    <KuiFormControlLayout
```

Comment:
I'd characterize the difference between the two approaches as mostly philosophical, which at first can seem arbitrary, though in my experience the result in two very different maintenance scenarios.

To put on my engineer hat, we're embracing the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) (SRP) by splitting up our form components into three categories: controls, layout, and row.

To adhere to SRP we want the single concern of each module to be clear to the user, and this concern should also contrast clearly against the concerns of other modules. Controls only surface the interactive parts of the form, layout decorates controls with other details like icons, and row coordinates how multiple {controls|layouts} appear together.

I think SRP has also created a set of components which compose well together, but don't depend on one another. For example, row presents controls well, but it can present _anything_ well. We can toss anything in there and it will never break because it has no dependency upon what it presents. And controls are presented well by rows, but they have no dependency upon rows. We can make changes to one without really having to worry about it will affect the other, which is great for maintenance.

Code isn't maintainable without being understandable, and clear responsibilities _should_ make this code easier to read and understand. Fewer responsibilities means fewer states and edge cases someone has to consider when making changes to code. For example, a `<KuiFieldSearch />` which sometimes renders a `<KuiFieldRow />` and sometimes renders a `<KuiIcon />` means having to hold all of those different states in your head and consider how your change will affect each one.

I'm not sure if I'm answering your question well... let me know how I did! 😄